IpmiFeaturePkg/GenericIpmi/Smm/SmmGenericIpmi.c

```diff
@@ -149,56 +62,55 @@ Routine Description:
     DEBUG ((EFI_D_ERROR, "ERROR!! Null Pointer returned by AllocateZeroPool ()\n"));
     ASSERT_EFI_ERROR (EFI_OUT_OF_RESOURCES);
     return EFI_OUT_OF_RESOURCES;
-  } else {
-    //
-    // Initialize the KCS transaction timeout. Assume delay unit is 1000 us.
-    //
-    mIpmiInstance->IpmiTimeoutPeriod = (PcdGet8 (PcdIpmiCommandTimeoutSeconds) * 1000*1000) / IPMI_DELAY_UNIT;
+  }
 
-    //
-    // Initialize IPMI IO Base, we still use SMS IO base to get device ID and Seltest result since SMM IF may have different cmds supported
-    //
-    mIpmiInstance->Signature                       = SM_IPMI_BMC_SIGNATURE;
-    mIpmiInstance->SlaveAddress                    = BMC_SLAVE_ADDRESS;
-    mIpmiInstance->BmcStatus                       = BMC_NOTREADY;
-    mIpmiInstance->IpmiTransport.IpmiSubmitCommand = IpmiSendCommand;
-    mIpmiInstance->IpmiTransport.GetBmcStatus      = IpmiGetBmcStatus;
+  //
+  // Initialize the KCS transaction timeout. Assume delay unit is 1000 us.
+  //
+  mIpmiInstance->IpmiTimeoutPeriod =
+    (PcdGet8 (PcdIpmiCommandTimeoutSeconds) * 1000*1000) / IPMI_DELAY_UNIT;

```

Comment:
nit: Spaces around operator (I thought Uncrustify would flag, but maybe it doesn't care)